lib/include/jxl/decode.h

```diff
@@ -550,6 +550,44 @@ JXL_EXPORT JxlDecoderStatus JxlDecoderGetColorAsICCProfile(
     const JxlDecoder* dec, const JxlPixelFormat* format,
     JxlColorProfileTarget target, uint8_t* icc_profile, size_t size);
 
+/** Sets the color profile to use for JXL_COLOR_PROFILE_TARGET_DATA for the
+ * special case when the decoder has a choice. This only has effect for a JXL
+ * image where uses_original_profile is false, and the original color profile is
+ * encoded as an ICC color profile rather than a JxlColorEncoding with known
+ * enum values. In most other cases (uses uses_original_profile is true, or the
+ * color profile is already given as a JxlColorEncoding), this setting is
+ * ignored and the decoder uses a profile related to the image.
+ * No matter what, the JXL_COLOR_PROFILE_TARGET_DATA must still be queried to
+ * know the actual data format of the decoded pixels after decoding.
+ *
+ * The intended use case of this function is for cases where you are using
+ * a color management system to parse the original ICC color profile
+ * (JXL_COLOR_PROFILE_TARGET_ORIGINAL), from this you know that the ICC
+ * profile represents one of the color profiles supported by JxlColorEncoding
+ * (such as sRGB, PQ or HLG): in that case it is beneficial (but not necessary)
+ * to use JxlDecoderSetDefaultColorProfile to match the parsed profile. The
+ * JXL decoder has no color management system built in, but can convert XYB
+ * color to any of the ones supported by JxlColorEncoding.
```

Comment:
Another potential use case of this would be for viewers to ignore the original color profile (enum or ICC) and try to get the image decoded immediately to the display space when possible, right?

E.g. when the display is Display P3 and the image has whatever original ICC profile (sRGB, P3, ProPhoto, whatever), then you may want to call `JxlDecoderSetPreferredColorProfile` with a Display P3 `color_encoding` so if it's an XYB image you directly get pixels in Display P3 (in other cases you still have to do your own color management).

Might be worth adding that to the comment.

Reply:
This setting doesn't *force* it however, if the decoder understands the color type (because it's an enum instead of ICC for XYB color) or cannot convert the color (because it's not XYB color but uses original profile), it ignores your setting.

The reason why it will ignore your setting if it's XYB color with enums (even though it theoretically could convert the XYB to your preferred choice) is that for blending (and some other operations, as far as I know), the specification requires the decoder to do it in that color space, so it would have to convert twice then. But in case of ICC profile with XYB, this does not apply, and decoder can decode to your preferred one.

A force option in theory is possible, but not for this MR :)